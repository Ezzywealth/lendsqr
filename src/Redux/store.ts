import { configureStore } from '@reduxjs/toolkit';
import counterReducer from './slices/userSlice';
import { useDispatch } from 'react-redux';

const store = configureStore({
	reducer: {
		counter: counterReducer,
	},
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export const useAppDispatch: () => AppDispatch = useDispatch;
export default store;
