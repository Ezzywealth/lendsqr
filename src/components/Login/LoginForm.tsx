import React from 'react';
import LoginHook from './LoginHook';
import { useSelector } from 'react-redux';
import { RootState } from '../../Redux/slices/authSlice';
import { CirclesWithBar, RotatingLines } from 'react-loader-spinner';
import LineWaveLoader from '../Shared/LineWaveLoader';

const LoginForm = () => {
	const { showPassword, setShowPassword, handleLogin, email, setEmail, password, setPassword } = LoginHook();
	const { loginLoading } = useSelector((state: RootState) => state.auth);

	return (
		<form className='login_form' onSubmit={handleLogin}>
			<section>
				<h2 className='form_title'>Welcome</h2>
				<p className='form_subtitle'>Enter details to login</p>
				<input disabled={loginLoading} type='text' placeholder='Email' className='input' value={email} onChange={(e) => setEmail(e.target.value)} />
				<div className='password_container'>
					<input disabled={loginLoading} type={showPassword ? 'text' : 'password'} placeholder='Password' className='input' value={password} onChange={(e) => setPassword(e.target.value)} />
					<button type='button' className='show_hide'>
						{showPassword ? (
							<p className='' onClick={() => setShowPassword(!showPassword)}>
								hide
							</p>
						) : (
							<p className='' onClick={() => setShowPassword(!showPassword)}>
								show
							</p>
						)}
					</button>
				</div>
				<div className='forgot_password'>
					<a href='#'>Forgot Password?</a>
				</div>
				<button type='submit' className='submit_btn' disabled={loginLoading}>
					{loginLoading ? (
						<div className='login_spinner'>
							<RotatingLines strokeColor='#fff' strokeWidth='5' animationDuration='0.75' width='20' visible={true} />
						</div>
					) : (
						'Login'
					)}
				</button>
			</section>
		</form>
	);
};

export default LoginForm;
