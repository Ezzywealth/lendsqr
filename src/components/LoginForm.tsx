import React, { useState } from 'react';

const LoginForm = () => {
	const [togglePassword, setTogglePassword] = useState(false);
	return (
		<form className='login-form'>
			<section>
				<h2 className='form_title'>Welcome</h2>
				<p className='form_subtitle'>Enter details to login</p>
				<input type='text' placeholder='Email' className='input' />
				<div className='password-container'>
					<input type='password' placeholder='Password' className='input' />
					<button type='button' className='show-hide'>
						{togglePassword ? (
							<p className='fas fa-eye-slash' onClick={() => setTogglePassword(!togglePassword)}>
								hide
							</p>
						) : (
							<p className='fas fa-eye' onClick={() => setTogglePassword(!togglePassword)}>
								show
							</p>
						)}
					</button>
				</div>

				<div className='forgot-password'>
					<a href=''>Forgot Password?</a>
				</div>

				<button type='submit' className='submit_btn'>
					Login
				</button>
			</section>
		</form>
	);
};

export default LoginForm;
