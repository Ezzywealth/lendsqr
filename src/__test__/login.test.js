/* eslint-disable testing-library/no-unnecessary-act */
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import LoginForm from '../components/Login/LoginForm';
import store from '../Redux/store';

describe('LoginForm', () => {
	test('renders LoginForm component', () => {
		const { container } = render(
			<Provider store={store}>
				<BrowserRouter>
					<LoginForm />
				</BrowserRouter>
			</Provider>
		);
		expect(container).toMatchSnapshot();
	});

	it('renders LoginForm correctly', () => {
		render(
			<Provider store={store}>
				<BrowserRouter>
					<LoginForm />
				</BrowserRouter>
			</Provider>
		);

		// Add assertions based on your component's structure and content
		expect(screen.getByText('Welcome')).toBeInTheDocument();
		expect(screen.getByPlaceholderText('Email')).toBeInTheDocument();
		expect(screen.getByPlaceholderText('Password')).toBeInTheDocument();
	});
});
