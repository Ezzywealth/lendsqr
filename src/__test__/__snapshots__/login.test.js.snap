// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders LoginForm component 1`] = `
<div>
  <form
    class="login-form"
  >
    <section>
      <h2
        class="form_title"
      >
        Welcome
      </h2>
      <p
        class="form_subtitle"
      >
        Enter details to login
      </p>
      <input
        class="input"
        placeholder="Email"
        type="text"
        value=""
      />
      <div
        class="password-container"
      >
        <input
          class="input"
          placeholder="Password"
          type="password"
          value=""
        />
        <button
          class="show-hide"
          type="button"
        >
          <p
            class=""
          >
            show
          </p>
        </button>
      </div>
      <div
        class="forgot-password"
      >
        <a
          href=""
        >
          Forgot Password?
        </a>
      </div>
      <button
        class="submit_btn"
        type="submit"
      >
        Login
      </button>
    </section>
  </form>
</div>
`;
